src/nat/front_ends/fastapi/job_store.py

```
@@ -304,24 +289,23 @@ async def submit_job(self,
-                    return (job_id, job)
-                except TimeoutError:
-                    pass
+        # Store the future in a variable, this allows us to potentially cancel the future later if needed
+        future_var = Variable(name=job_id, client=self.dask_client)
+        future_var.set(future, timeout="5 s")
+        if sync_timeout > 0:
+            try:
+                future.result(timeout=sync_timeout)
+                job = await self.get_job(job_id)
+                assert job is not None, "Job should exist after future result"
+                return (job_id, job)
```

Ruff (0.14.14)

Consider moving this statement to an `else` block

(TRY300)